arch/risc-v/src/bl808/bl808_gpio.c

```diff
@@ -0,0 +1,148 @@
+/****************************************************************************
+ * arch/risc-v/src/bl808/bl808_gpio.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <debug.h>
+#include <sys/param.h>
+
+#include "riscv_internal.h"
+#include "hardware/bl808_glb.h"
+#include "bl808_gpio.h"
+
+////TODO
+////#define BL808_GLB_BASE        0x20000000ul  /* glb */
+#define BL808_GPIO_BASE 0x200008c4ul /* gpio */
+#define BL808_NGPIOS 45
+#define reg_gpio_xx_o 24
+#define reg_gpio_xx_i 28
```

Comment:
Please rename this and move to arch/risc-v/src/bl808/bl808_gpio.h

```suggestion
#define GPIO_I_SHIFT 28
```

Reply:
We might need to change line 141:
```
  return (regval & (1 << reg_gpio_xx_i)) != 0;
```
```
  return (regval & (1 << GPIO_I_SHIFT)) != 0;
```

Reply:
Correct please make the changes so it will compile

Reply:
Okay